src/OrbitBase/JoinFutures.cpp

```diff
@@ -42,11 +42,19 @@ Future<void> JoinFutures(absl::Span<const Future<void>> futures) {
           }
         });
 
-    if (result == orbit_base::FutureRegisterContinuationResult::kSuccessfullyRegistered) {
-      ++shared_state->incompleted_futures;
+    if (result != orbit_base::FutureRegisterContinuationResult::kSuccessfullyRegistered) {
```

Comment:
Is this use-case testable? Maybe add a unit-test for this?

Reply:
Yes, can be tested. I will add a test!

Reply:
Done.